resolver/service.go

```diff
@@ -144,7 +144,12 @@ func (s *Service) UseServices() []*Service {
 	return services
 }
 
-func (s *Service) useServices(resolver *MessageResolver) []*Service {
+func (s *Service) useServices(resolver *MessageResolver, msgResolverMap map[*MessageResolver]struct{}) []*Service {
```

Comment:
Slightly updated from https://github.com/mercari/grpc-federation/pull/16 and passed `msgResolverMap` to the `useServices` method directly so that we don't have to define the `useServicesRecursive` method.